src/platform/forms-system/src/js/helpers.js

```diff
@@ -298,45 +298,83 @@ function isPartialAddress(data) {
+ * Replaces non-escaping backslashes with forward-slashes.
+ *
+ * @param {string} string
+ * @returns {string}
+ */
+export const replaceEscapedCharacters = string => {
+  return string
+    .replaceAll('"', "'")
+    .replace(/(?:\r\n|\n\n|\r|\n)/g, '; ')
+    .replace(/(?:\t|\f|\b)/g, '')
+    .replace(/\\(?!(f|n|r|t|[u,U][\d,a-fA-F]{4}))/gm, '/');
+};
+
 /**
  * Create a replacer function for JSON.stringify
  * A replacer function is the second argument to JSON.stringify
  *
- * @param {{
- *  allowPartialAddress?: boolean,
- * }} [options] - Options for the replacer
+ * @param {ReplacerOptions} [options] - Options for the replacer
+ * @returns {(key, value) => any} The replacer function
  */
 export function createStringifyFormReplacer(options) {
+  /**
+   * The replacer function
+   *
+   * @param {*} key
+   * @param {*} value
+   * @returns {*} Replaced value
+   */
   const replacerFn = (key, value) => {
-    if (!options?.allowPartialAddress && isPartialAddress(value)) {
-      return undefined;
-    }
+    /**
+     * Replaces objects (including arrays) with other values
+     *
+     * @param {Object} object
+     * @returns {*} The replacement value
+     */
+    const replaceObject = object => {
+      // Clean up empty objects in arrays
+      if (Array.isArray(value)) {
+        const newValues = value.filter(v => !!replacerFn(key, v));
+        // If every item in the array is cleared, remove the whole array
+        return newValues.length > 0 ? newValues : undefined;
+      }
 
-    if (typeof value === 'object') {
-      const fields = Object.keys(value);
+      if (!options?.allowPartialAddress && isPartialAddress(value)) {
+        return undefined;
+      }
+
+      const fields = Object.keys(object);
       if (
         fields.length === 0 ||
-        fields.every(field => value[field] === undefined)
+        fields.every(field => object[field] === undefined)
       ) {
         return undefined;
       }
 
       // autosuggest widgets save value and label info, but we should just return the value
-      if (value.widget === 'autosuggest') {
-        return value.id;
+      if (object.widget === 'autosuggest') {
+        return object.id;
       }
 
       // Exclude file data
-      if (value.confirmationCode && value.file) {
-        return omit('file', value);
+      if (object.confirmationCode && object.file) {
+        return omit('file', object);
       }
-    }
 
-    // Clean up empty objects in arrays
-    if (Array.isArray(value)) {
-      const newValues = value.filter(v => !!replacerFn(key, v));
-      // If every item in the array is cleared, remove the whole array
-      return newValues.length > 0 ? newValues : undefined;
+      return object;
+    };
+
+    if (typeof value === 'object') {
+      return replaceObject(value);
+    }
+    if (typeof value === 'string') {
+      return options?.replaceEscapedCharacters
+        ? replaceEscapedCharacters(value)
+        : value;
```

Comment:
**QUESTION**
This feels like the meat of the change here. What was happening before this change? Strings were not undergoing any of this escaped-character processing? Was that problematic? Is there some specific reason we need this while previous transformations could do without it?

Reply:
> What was happening before this change?
You're correct. Strings were not undergoing any escaped-character processing. Do we need it? Maybe, maybe not. This logic was originally added to the shared simple forms transformer in #24965 to fix a bug where the JSON payload would break if `"` or `\` was in the submission data. I would imagine our forms would be vulnerable to the same bug any place we implement a text input, so out of an abundance of caution, I decide to add it as an option and enable the option for all Form Engine forms. I'm open to changing direction if you think we do not need it or if the VFF team feels strongly it should be enabled in all forms by default.